### PR TITLE
RFQ refactor towards adding asset sell request support

### PR DESCRIPTION
**Pull request by ffranr**
Towards fixing https://github.com/lightninglabs/taproot-assets/issues/818

This PR contains commits which refactor the RFQ service towards solving https://github.com/lightninglabs/taproot-assets/issues/818 .

The changes in this PR shouldn't change the functionality of the RFQ service. They should rename methods, functions, types, and variables so that the changes necessary for https://github.com/lightninglabs/taproot-assets/issues/818 are easier to make in a different PR.

### CODE REVIEW

File: rpcserver.go [hunk 1]
```diff
@@ -4999,18 +5001,20 @@ func marshalAcceptedQuotes(
 	return rpcQuotes
 }
 
-// QueryRfqAcceptedQuotes queries for accepted quotes from the RFQ system.
-func (r *rpcServer) QueryRfqAcceptedQuotes(_ context.Context,
-	_ *rfqrpc.QueryRfqAcceptedQuotesRequest) (
-	*rfqrpc.QueryRfqAcceptedQuotesResponse, error) {
+// QueryRfqPeerAcceptedQuotes is used to query for quotes that were requested by
+// our node and have been accepted our peers.
+func (r *rpcServer) QueryRfqPeerAcceptedQuotes(_ context.Context,
```
Comment (guggero) [inline rpcserver.go:5006]: nit: since we're renaming: This is located in the `rfqrpc` package, within the `Rfq` service. Do we really need the `Rfq` in this name as well? I think we can shorten some RPC names if we consider that there's always the package name context to help.

File: taprpc/rfqrpc/rfq.proto [hunk 2]
```diff
@@ -106,19 +106,21 @@ message AcceptedQuote {
     uint64 expiry = 6;
 }
 
-message QueryRfqAcceptedQuotesResponse {
-    repeated AcceptedQuote accepted_quotes = 1;
+message QueryRfqPeerAcceptedQuotesResponse {
```
Comment (guggero) [inline taprpc/rfqrpc/rfq.proto:109]: Same here, IMO we can remove the `Rfq` part from all message names.

File: rfqmsg/messages.go [hunk 3]
```diff
@@ -17,8 +21,15 @@ func (id ID) String() string {
 	return hex.EncodeToString(id[:])
 }
 
-// SerialisedScid is a serialised short channel id (SCID).
-type SerialisedScid uint64
+func (id ID) Scid() SerialisedScid {
```
Comment (guggero) [inline rfqmsg/messages.go:24]: nit: missing Godoc.

File: chain_bridge.go [hunk 4]
```diff
@@ -87,13 +87,13 @@ func (l *LndRpcChainBridge) GetBlock(ctx context.Context,
 func (l *LndRpcChainBridge) GetBlockHeader(ctx context.Context,
 	hash chainhash.Hash) (*wire.BlockHeader, error) {
 
-	header, err := l.lnd.ChainKit.GetBlockHeader(ctx, hash)
+	block, err := l.lnd.ChainKit.GetBlock(ctx, hash)
```
Comment (guggero) [inline chain_bridge.go:90]: Why is this change needed? This is actually a performance regression, because fetching the full block is always more expensive than fetching the header (especially for Neutrino).

**Review by guggero — CHANGES_REQUESTED** [review]
Changes look good with the exception of the last commit (see inline comment).

File: chain_bridge.go (hunk 4 above)
Comment (ffranr) [reply chain_bridge.go]: I've removed this change, thanks for pointing it out. I think I made this change because I was originally using a local lndclient repo.

File: rpcserver.go (hunk 1 above)
Comment (ffranr) [reply rpcserver.go:5006]: I'll shorten the name.

I included the RFQ package name because I wanted to make sure that the method would be unique in the `rpcServer` struct method namespace also.

Comment (guggero) [reply rpcserver.go:5006]: That makes sense. Though I think if we ever start to experience naming collisions in the RPC server, then it's time to split that code up into several structs to mirror the packages they implement.

File: rfqmsg/messages.go [hunk 5]
```diff
@@ -17,8 +21,16 @@ func (id ID) String() string {
 	return hex.EncodeToString(id[:])
 }
 
-// SerialisedScid is a serialised short channel id (SCID).
-type SerialisedScid uint64
+// Scid returns the short channel id (SCID) of the RFQ message.
+func (id ID) Scid() SerialisedScid {
+	// Given valid RFQ message id, we then define a RFQ short chain id
```
Comment (GeorgeTsagk) [inline rfqmsg/messages.go:26]:
```suggestion
	// Given valid RFQ message id, we then define a RFQ short channel id
```

File: rfqmsg/messages.go [hunk 6]
```diff
@@ -17,8 +21,16 @@ func (id ID) String() string {
...
+	// (SCID) by taking the last 8 bytes of the RFQ message id and
+	// interpreting them as a 64-bit integer.
+	scidBytes := id[24:]
```
Comment (GeorgeTsagk) [inline rfqmsg/messages.go:29]: nit
```suggestion
	scidBytes := id[len(id)-8]
```

**Review by GeorgeTsagk — APPROVED** [review]
LGTM :zap: 
few nits

File: rfqmsg/messages.go (hunk 6 above)
Comment (ffranr) [reply rfqmsg/messages.go:29]: I don't think your suggesting works. We want to select the last 8 elements from the `id` array, not the single element at `len(id)-8`.

**Comment by ffranr** [comment]
Last commit (`lndclient` version bump) wasn't necessary after rebase.
